apps/passport-client/components/screens/ZappScreens/ZappScreen.tsx

```diff
@@ -0,0 +1,36 @@
+import { ReactNode } from "react";
+import { useDispatch, useEmbeddedScreenState } from "../../../src/appHooks";
+import { ListenMode, useZappServer } from "../../../src/zapp/useZappServer";
+import { AdhocModal } from "../../modals/AdhocModal";
+import { EmbeddedScreen } from "../EmbeddedScreens/EmbeddedScreen";
+
+export function ZappScreen({ url }: { url: string }): ReactNode {
+  return (
+    <>
+      <ZappModal />
+      <iframe
```

Comment:
if it doesn't already, can we have this preload the webpage in the background so that it shows up quickly, if not immediately, once the button is pressed?

https://stackoverflow.com/questions/32547844/how-to-preload-iframes

Reply:
The recommended answer in that thread has a few people saying that it doesn't work, and [MDN doesn't list "document" as a supported "as" attribute for <link>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link). Probably the only reliable option is to load the iframe in the background, but this means that it's sitting there consuming resources, running scripts, making network requests, and so on, even if the user never interacts with it.

Reply:
We could make the loading process visually smoother by hiding the iframe until the Zapp is fully loaded, showing a nicer loading screen in the meantime?

Reply:
I've made a new Linear issue for this.